helics/cli.py

```diff
@@ -305,8 +306,14 @@ def run(path, silent, connect_server, no_log_files, no_kill_on_error):
             if "env" in f:
                 for k, v in f["env"].items():
                     env[k] = v
+            p_args = shlex.split(f["exec"])
+            p_args[0] = shutil.which(p_args[0])
+            if p_args[0] is None:
+                raise click.ClickException("UnrecognizedCommandError: The command specified in exec string is not a "
```

Comment:
If the output with the exception makes it clear enough what the cause is, then the exception seems fine.

The alternative to raising an exception through click is calling the `error` function (which... also uses click) and then `exit` with an error code (?) -- calling `error` seems a bit more common in this file, but there is another instance of raising an exception in this same run subcommand.